src/migrations/Migrator.js

```diff
@@ -51,22 +52,43 @@ class Migrator {
     }
   }
 
-  async initMigrations() {
+  async waitForOpenWindowAndStartMigration(logger) {
+    const sleep = async (timeout) => new Promise((resolve) => {
+      setTimeout(resolve, timeout * 1000);
+    });
+    let migrationStartTime;
+    do {
+      const {modifiedCount} = await this.db.collection('migrations').updateOne({migrationStartTime: {$exists: false}}, {$set: {migrationStartTime: getTimestamp()}});
+      if (modifiedCount > 0) {
+        return;
+      }
+      logger.info('Another migration is running. Waiting for it to end');
+      ({migrationStartTime} = (await this.db.collection('migrations').findOne({})));
+      await sleep(this.config.migrationSleepTime);
+    } while (migrationStartTime + this.config.migrationTaskTimeout > getTimestamp());
+  }
+
+  async initMigrations(logger) {
     await this.db.createCollection('migrations');
     const result = await this.db.collection('migrations').findOne({});
```

Comment:
As much of this as possible should by covered by the mutex

Reply:
👍

Reply:
Could not cover it by mutex (creates chicken-egg problem), so just collection creation to be done atomically